GVFS/GVFS.Common/FileSystem/PhysicalFileSystem.cs

```diff
@@ -27,7 +27,10 @@ public virtual void DeleteDirectory(string path, bool recursive = true, bool ign
                 foreach (FileInfo file in directory.GetFiles())
                 {
                     file.Attributes = FileAttributes.Normal;
-                    file.Delete();
+                    if(file.Exists)

```

Comment:
there is still a race here that the file exists and is then deleted before the delete is invoked; a better solution is to perform the delete unconditionally and then catch the exception if the file does not exist